lib/compiler-cranelift/src/config.rs

```diff
@@ -186,6 +186,11 @@ impl Cranelift {
 
         settings::Flags::new(flags)
     }
+
+    /// Get Middlewares
+    pub fn get_middlewares(&self) -> Vec<Arc<dyn ModuleMiddleware>> {
+        self.middlewares.clone()
+    }
```

Comment:
Consider using `&[Arc<dyn ModuleMiddleware>]` it won't clone every `Arc` or allocate a new container when this is called. Callers can then get vecs of it with `.to_vec()`.

The primary reason to use `Vec` over `&[]` in a trait method is that you don't want to impose on the implementors that they should actually be storing the data that they're returning. `Vec` lets you just build it and return it, `&[]` requires that the data actually lives on the thing the method is being called on (and that the data is stored in such a way that you can return a `&[]` to it, i.e. linear storage like a Vec or an array).

Reply:
Fixed!